portability-worker/src/main/java/org/dataportabilityproject/worker/JobPollingService.java

```diff
@@ -40,124 +42,126 @@
-        jobId,
-        publicKey.getEncoded().length);
-  }
 
-  /**
-   * Claims {@link PortabilityJob} {@code jobId} and updates it with our public key in storage.
-   */
-  private void claimJob(UUID jobId, KeyPair keyPair) throws IOException {
-    // Lookup the job so we can append to its existing properties.
-    PortabilityJob existingJob = store.findJob(jobId);
-    // Verify no worker key
-    if (existingJob.jobAuthorization().encodedPublicKey() != null) {
-      throw new IOException("public key cannot be persisted again");
+    /**
+     * Polls for an unassigned job, and once found, initializes the global singleton job metadata
+     * object for this running instance of the worker.
+     */
+    private void pollForUnassignedJob() throws IOException {
+        UUID jobId = store.findFirst(JobAuthorization.State.CREDS_AVAILABLE);
+        logger.debug("Polling for a job CREDS_AVAILABLE");
+        if (jobId == null) {
+            return;
+        }
+        logger.debug("Polled job {}", jobId);
+        Preconditions.checkState(!JobMetadata.isInitialized());
+        KeyPair keyPair = asymmetricKeyGenerator.generate();
+        PublicKey publicKey = keyPair.getPublic();
+        // TODO: Back up private key (keyPair.getPrivate()) in case this worker dies mid-copy, so we
+        // don't have to make the user start from scratch. Some options are to manage this key pair
+        // within our hosting platform's key management system rather than generating here, or to
+        // encrypt and store the private key on the client.
+        // Note: claimJob may fail if another worker beat us to it. That's ok -- this worker will keep
+        // polling until it can claim a job.
+        claimJob(jobId, keyPair);
+        logger.debug(
+                "Updated job {} to CREDS_ENCRYPTION_KEY_GENERATED, publicKey length: {}",
+                jobId,
+                publicKey.getEncoded().length);
     }
-    // Populate job with public key to persist
-    String encodedPublicKey = BaseEncoding.base64Url().encode(keyPair.getPublic().getEncoded());
 
-    PortabilityJob updatedJob =
-        existingJob
-            .toBuilder()
-            .setAndValidateJobAuthorization(
+    /**
+     * Claims {@link PortabilityJob} {@code jobId} and updates it with our public key in storage.
+     */
+    private void claimJob(UUID jobId, KeyPair keyPair) throws IOException {
+        // Lookup the job so we can append to its existing properties.
+        PortabilityJob existingJob = store.findJob(jobId);
+        // Verify no worker key
+        if (existingJob.jobAuthorization().encodedPublicKey() != null) {
+            throw new IOException("public key cannot be persisted again");
+        }
+        // Populate job with public key to persist
+        String encodedPublicKey = BaseEncoding.base64Url().encode(keyPair.getPublic().getEncoded());
+
+        PortabilityJob updatedJob =
```

Comment:
Hmm this clause is formatted a bit funny. do you think we can make it cleaner by moving the subclause in setAndValidatejobAuthorization a separate clause?